From d8cce6388b16d559f541616986356be5347d1cca Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Wed, 18 Jul 2018 19:56:43 +0800
Subject: [PATCH 4278/5242] MLK-18969 ARM64: dts: fsl-imx8mm-evk: add touch
 support

commit  8929937b3625dc37507726fd0a3bd2ff9d037f8f from
https://source.codeaurora.org/external/imx/linux-imx.git

MIPI DSI panel also has a touch, this patch add the touch
support for fsl-imx8mm-evk board.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
(cherry picked from commit 8246c3148e07eeda05b5b375819c06bde0a80824)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts  |   12 ++++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts   |    6 ++++++
 2 files changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts
index 1241b9f3..06488ae 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk-rm67191.dts
@@ -20,3 +20,15 @@
 		status = "okay";
 	};
 };
+
+&i2c2 {
+	synaptics_dsx_ts@20 {
+		compatible = "synaptics_dsx";
+		reg = <0x20>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_i2c2_synaptics_dsx_io>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <9 IRQ_TYPE_LEVEL_LOW>;
+		status = "okay";
+	};
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
index 912350e..08ebdc9 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -166,6 +166,12 @@
 			>;
 		};
 
+		pinctrl_i2c2_synaptics_dsx_io: synaptics_dsx_iogrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_GPIO1_IO09_GPIO1_IO9		0x19	/* Touch int */
+			>;
+		};
+
 		pinctrl_fec1: fec1grp {
 			fsl,pins = <
 				MX8MM_IOMUXC_ENET_MDC_ENET1_MDC		0x3
-- 
1.7.9.5

